transports/noise/src/io/handshake.rs

```diff
@@ -207,8 +207,8 @@ where
     }
     let pb = pb_result?;
 
-    if !pb.identity_key.is_empty() {
-        let pk = identity::PublicKey::from_protobuf_encoding(&pb.identity_key)
+    if !pb.identity_key.as_ref().expect("optional").is_empty() {
```

Comment:
We can't just panic here :)

We need more graceful error handling to check whether or not the key is actually present. Probably best done with a `match` and a match-arm guard so you can do the existing `if` and the destructing in one go.

Have a google around if some of these terms don't mean anything to you (yet)! :)